zds/utils/templatetags/nb_word_to_time.py

```diff
@@ -0,0 +1,25 @@
+# coding: utf-8
+
+from django import template
+from zds.settings import ZDS_APP
+register = template.Library()
+
+
+@register.filter(name="nb_word_to_icontime")
+def nb_word_to_icontime(nb_word):
+    """
+    Gets the icon for this time to read the content
+    :param nb_word:
+    :return:
+    """
+    association = ["icon-15-min", "icon-30-min", "icon-45-min", "icon-60-min", "icon-more-min"]
```

Comment:
En fait quand je dis que ça "complexifie le code" c'est juste que ça 
force à jouer avec des if/elif au lieu d'un simple trick sur les list.


Le 22/06/2016 à 21:32, Pierre Beaujean a écrit :
>
> In zds/utils/templatetags/nb_word_to_time.py 
> <https://github.com/zestedesavoir/zds-site/pull/3681#discussion_r68119428>:
>
> > @@ -0,0 +1,25 @@
> > +# coding: utf-8
> > +
> > +from django import template
> > +from zds.settings import ZDS_APP
> > +register = template.Library()
> > +
> > +
> > +@register.filter(name="nb_word_to_icontime")
> > +def nb_word_to_icontime(nb_word):
> > +    """
> > +    Gets the icon for this time to read the content
> > +    :param nb_word:
> > +    :return:
> > +    """
> > +    association = ["icon-15-min", "icon-30-min", "icon-45-min", "icon-60-min", "icon-more-min"]
>
> Chui pour le logarithme, qui apportera une échelle relativement plus 
> logique (après, on peut argumenter sur le fait qu'une journée de 24 
> heures, c'est irréaliste en pratique). En soit, c'est juste 
> |math.log()| et à part les cas tordus, c'est un nombre positif ;)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/zestedesavoir/zds-site/pull/3681/files/13758fdcf27f091fc54b1ffb111309ee52aeef78#r68119428>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ABc_xfRufQLPcj-54iEofMjN3nnT3Labks5qOY2xgaJpZM4I7sFO>.
>